lib/matplotlib/_type1font.py

```diff
@@ -35,10 +37,64 @@
 
 from matplotlib.cbook import _format_approx
 from . import _api
+if T.TYPE_CHECKING:
+    from collections.abc import Iterable
 
 _log = logging.getLogger(__name__)
 
 
+class _Tagger:
+    __slots__ = ('_tag', '_next')
+    _tag: dict[frozenset[str], str]
+    _next: str
+
+    def __init__(self):
+        self._tag = {}
+        self._next = 'MPLAAA'
+
+    def __call__(self, chars: Iterable[str]) -> str:
+        """
+        Return a six-character tag made of uppercase letters and a plus sign
+
+        Useful for adding a tag into subsetted fonts while keeping the code
+        reproducible. The function always returns the same value for the
+        same set on the same exact Python version and is free of collisions.
```

Comment:
"always returns the same value..." -> if the sequence of calls is the same.

Reply:
Yeah, and if we want reproducible output we shouldn't really make this a global variable. I suppose the tagger should be a part of the PDF file object so that it can ensure that tags within the same file are consistent.